packages/runtime-dom/src/components/Transition.ts

```diff
@@ -194,7 +194,7 @@ export function resolveTransitionProps(
       callHook(hook, [el, resolve])
       nextFrame(() => {
         removeTransitionClass(el, isAppear ? appearFromClass : enterFromClass)
-        if (__COMPAT__ && legacyClassEnabled) {
+        if (__COMPAT__ && legacyClassEnabled && legacyAppearFromClass && legacyEnterFromClass) {
```

Comment:
Can't verify on the spot as this PR doesn' contain a test case (no judgement, I am unsure myself where to put such a test right now), but:

wouldn't a more straightforward solution be to just initialize the `legacy*`variables as empty strings (see lines 143-145)